go/chat/bots/commands.go

```diff
@@ -0,0 +1,366 @@
+package bots
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+
+	"github.com/keybase/client/go/protocol/keybase1"
+
+	"github.com/keybase/client/go/encrypteddb"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/storage"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+)
+
+type commandUpdaterJob struct {
+	convID     chat1.ConversationID
+	info       *chat1.BotInfo
+	completeCh chan error
+}
+
+type userCommandAdvertisement struct {
+	Alias    *string                     `json:"alias,omitempty"`
+	Commands []chat1.UserBotCommandInput `json:"commands"`
+}
+
+type storageCommandAdvertisement struct {
+	Advertisement userCommandAdvertisement
+	Username      string
+}
+
+type commandsStorage struct {
+	Advertisements []storageCommandAdvertisement `codec:"A"`
+}
+
+const commandsStorageVersion = 1
+
+var commandsPublicTopicName = "___keybase_botcommands_public"
```

Comment:
`const`?

Reply:
I take it's address at some point so I made it this.